File: ws/santorum/u1/ws06Json/Model/ChickenFarmer.js
class ChickenFarmer {
  constructor(name) {
    this.name = name;
    this.coops = [];
  }

  addCoop(coop) {
    this.coops.push(coop);
  }

  getAllChickens() {
    return this.coops.flatMap(coop => coop.getChickens());
  }
}

File: hw/ramos/u1/hw14SafeInJsonChickenFarmSimulator/python/model/ChickenFarmer.py
class ChickenFarmer:
    def __init__(self, name: str):
        self.name = name
        self.coops = []

    def add_coop(self, coop):
        self.coops.append(coop)

    def get_all_chickens(self):
        # flatMap equivalente: recorrer todos los gallineros y juntar sus pollos
        all_chickens = []
        for coop in self.coops:
            all_chickens.extend(coop.get_chickens())
        return all_chickens
